wgsl/index.bs

```diff
@@ -865,135 +1031,47 @@ It is defined recursively in the following table:
+</p>
 
-We write |Extent|(|V|,|C|) for the allocation extent of value |V|
-of host-shareable type |S| in storage class |C|,
-where |C| is either [=storage classes/storage=] or [=storage classes/storage=].
-It is defined recursively in the following table:
+All arrays of element type |T| must have an element [=stride=] that is a
+multiple of [=RequiredAlignOf=](|T|, |C|) for the storage class |C|:
 
-<table class='data'>
-  <caption>Allocation extent of a value of host-shareable type</caption>
-  <thead>
-    <tr><th>Host-shareable type |S|
-        <th>|Extent|(|V|,`storage`)<br>where |V| is of type |S|
-        <th>|Extent|(|V|,`uniform`)<br>where |V| is of type |S|
-  </thead>
-  <tr><td>[=i32=], [=u32=], or [=f32=]
-      <td>4
-      <td>4
-  <tr><td>vec|N|&lt;|T|&gt;,
-      where |T| is one of [=i32=], [=u32=], or [=f32=]
-      <td>|N| &times; 4
-      <td>|N| &times; 4
-  <tr algorithm="extent of a matrix with 2 rows">
-      <td>mat|N|x2&lt;f32&gt;
-      <td>|N| &times; 8
-      <td>|N| &times; 8
-  <tr algorithm="extent of a matrix with 3 rows">
-      <td>mat|N|x3&lt;f32&gt;
-      <td>|N| &times; 16
-      <td>|N| &times; 16
-  <tr algorithm="extent of a matrix with 4 rows">
-      <td>mat|N|x4&lt;f32&gt;
-      <td>|N| &times; 16
-      <td>|N| &times; 16
-  <tr algorithm="extent of an array">
-      <td>array&lt;<var ignore>T</var>,|N|>
-      <td>|N| &times; |Stride|(|S|)
-      <td>|N| &times; |Stride|(|S|)
-  <tr algorithm="extent of an runtime-sized array">
-      <td>array&lt;<var ignore>T</var>&gt;
-      <td>|N|<sub>runtime</sub> &times; |Stride|(|S|),<br>
-where |N|<sub>runtime</sub> is the runtime-determined number of elements of <var ignore>V</var>
-      <td>Not applicable:  [=runtime-sized=] arrays cannot appear in [=storage classes/storage=] storage
-  <tr algorithm="extent of a structure">
-      <td>struct&lt;|T|<sub>1</sub>,...,|T|<sub>|n|</sub>&gt;
-      <td>[=roundUp=](|Align|(|S|,`storage`),|L|),<br>
-      where |L|&nbsp;=&nbsp;|Offset|(|S|,|n|)&nbsp;+&nbsp;|Extent|(|V|<sub>|n|</sub>,`storage`)),<br>
-      and |V|<sub>|n|</sub> is the last member of |V|
-      <td>[=roundUp=](|Align|(|S|,`uniform`),|L|),<br>
-      where |L|&nbsp;=&nbsp;|Offset|(|S|,|n|)&nbsp;+&nbsp;|Extent|(|V|<sub>|n|</sub>,`uniform`)),<br>
-      and |V|<sub>|n|</sub> is the last member of |V|
-</table>
+<p algorithm="array element minimum alignment">
+    [=StrideOf=](array<|T|[, |N|]>) = |k| &times; [=RequiredAlignOf=](|T|, C)<br>
+    Where |k| is a non-negative integer
+</p>
+
+
+The [=storage classes/uniform=] storage class also requires that:
 
-When a type |S| is not a runtime-sized array and it does not contain a runtime-sized array, then
-all values |V| of type |S| will have the same allocation extent for a storage class |C|.
-In these cases we define the allocation extent of the *type* |S| as that common value:
-|Extent|(|S|,|C|) = |Extent|(|V|,|C|), for any |V| of type |S|.
+* Array elements are aligned to 16 byte boundaries.
```

Comment:
I think these are redundant based on the rules above.

Reply:
I'm not sure?

Array *element* locations depend on the array member stride and the array element stride. So, a `array<i32>` *member* has a 16 byte alignment requirement (as based on the rules above), but the stride between elements also needs to be a multiple of 16 bytes.

Reply:
That's fair, maybe I was assuming StrideOf(array<T>, C) where C is uniform was be described as roundUp(RequiredAlignOf(T, C), 16), but looking now I don't see that. So which ever is simpler to describe is ok.